internal/services/compute/virtual_machine_scale_set.go

```diff
@@ -378,6 +378,46 @@ func FlattenVirtualMachineScaleSetSpotRestorePolicy(input *virtualmachinescalese
 	}
 }
 
+func ExpandVirtualMachineScaleSetResiliency(resilientVMCreationEnabled, resilientVMDeletionEnabled bool) *virtualmachinescalesets.ResiliencyPolicy {
+	// Note: AutomaticZoneRebalancingPolicy is excluded as it's in private preview and
+	// has been removed from the schema to prevent API errors.
+	result := &virtualmachinescalesets.ResiliencyPolicy{}
+
+	// Always include both policies to maintain consistent API behavior
+	result.ResilientVMCreationPolicy = &virtualmachinescalesets.ResilientVMCreationPolicy{
+		Enabled: pointer.To(resilientVMCreationEnabled),
+	}
+
+	result.ResilientVMDeletionPolicy = &virtualmachinescalesets.ResilientVMDeletionPolicy{
+		Enabled: pointer.To(resilientVMDeletionEnabled),
+	}
+
+	return result
+}
+
+func FlattenVirtualMachineScaleSetResiliency(input *virtualmachinescalesets.ResiliencyPolicy) (bool, bool) {
+	// Default values for the top-level fields
+	resilientVMCreationEnabled := false
+	resilientVMDeletionEnabled := false
+
+	if input == nil {
+		// No ResiliencyPolicy - don't set these fields in state for backward compatibility
+		return resilientVMCreationEnabled, resilientVMDeletionEnabled
+	}
+
+	// Extract top-level fields - these are always set in Terraform state based on Azure response
+	if vmCreation := input.ResilientVMCreationPolicy; vmCreation != nil && vmCreation.Enabled != nil {
+		resilientVMCreationEnabled = pointer.From(vmCreation.Enabled)
+	}
+
+	if vmDeletion := input.ResilientVMDeletionPolicy; vmDeletion != nil && vmDeletion.Enabled != nil {
+		resilientVMDeletionEnabled = pointer.From(vmDeletion.Enabled)
+	}
+
+	// ResiliencyPolicy exists - set fields in state
+	return resilientVMCreationEnabled, resilientVMDeletionEnabled
+}
```

Comment:
Could we adjust the implementation to instead use named return values? I am afraid using positional return values is too easy to mix up which is which.

```suggestion
func FlattenVirtualMachineScaleSetResiliency(input *virtualmachinescalesets.ResiliencyPolicy) (resilientVMCreationEnabled, resilientVMDeletionEnabled bool) {
    if input == nil {
        return
    }

    if vmCreation := input.ResilientVMCreationPolicy; vmCreation != nil {
        resilientVMCreationEnabled = pointer.From(vmCreation.Enabled)
    }

    if vmDeletion := input.ResilientVMDeletionPolicy; vmDeletion != nil {
        resilientVMDeletionEnabled = pointer.From(vmDeletion.Enabled)
    }

    return
```

Reply:
Fixed.